### PR TITLE
feat: LT-996 - Respond to ECJU queries

### PR DESCRIPTION
**Pull request by jfaracik**

### CODE REVIEW

File: applications/views.py [hunk 1]
```diff
@@ -150,22 +149,6 @@ def put(self, request, pk):
...
-        application = get_application_by_pk(pk)
-        request.user.notification_set.filter(note__case__application=application).update(
-            viewed_at=timezone.now()
-        )
-
-        return super(ApplicationDetailUser, self).get(request, pk)
-
-
 class CLCList(APIView):
     def post(self, request):
```
Comment (lukesa) [inline applications/views.py]: good moment to add docstring to document API method

File: cases/migrations/0004_auto_20190829_1049.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# Generated by Django 2.2.4 on 2019-08-29 10:49
```
Comment (lukesa) [inline cases/migrations/0004_auto_20190829_1049.py]: rename migration file

File: cases/tests/tests_case_ecju_queries.py [hunk 3]
```diff
@@ -20,25 +19,41 @@ def setUp(self):
 
         ecju_query = EcjuQuery(question='ECJU Query 1', case=self.case, raised_by_user=self.gov_user)
         ecju_query.save()
+
+        self.team_2 = self.create_team('TAU')
+        self.gov_user_2 = self.create_gov_user('bob@slob.com', self.team_2)
```
Comment (lukesa) [inline cases/tests/tests_case_ecju_queries.py]: move email into const: GOV_USER2_EMAIL

File: cases/views.py [hunk 4]
```diff
@@ -306,12 +313,33 @@ class EcjuQueryDetail(APIView):
...
     def get(self, request, pk, ecju_pk):
         """
         Returns details of an ecju query
         """
         ecju_query = get_ecju_query(ecju_pk)
-        serializer = EcjuQuerySerializer(ecju_query)
+        serializer = EcjuQueryExporterSerializer(ecju_query)
         return JsonResponse(data={'ecju_query': serializer.data})
+
+    def put(self, request, pk, ecju_pk):
```
Comment (lukesa) [inline cases/views.py]: docstring missing

File: cases/libraries/mark_notifications_as_viewed.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+from cases.models import CaseNote, Notification, EcjuQuery
+
+
+def mark_notifications_as_viewed(user, objects):
+    for obj in objects:
+        if isinstance(obj, CaseNote):
+            Notification.objects.filter(user=user, case_note=obj).update(
+                viewed_at=timezone.now())
+        elif isinstance(obj, EcjuQuery):
```
Comment (BenAndrew-Kainos) [inline cases/libraries/mark_notifications_as_viewed.py]: As both the if and elif do the the same task just use 'if isinstance(obj, CaseNote) or isinstance(obj, EcjuQuery):'

File: cases/models.py [hunk 6]
```diff
@@ -133,7 +128,31 @@ class EcjuQuery(models.Model):
...
     responded_by_user = models.ForeignKey(ExporterUser, related_name='exportuser_ecju_query', on_delete=models.CASCADE,
                                           default=None, null=True)
+
+    # pylint: disable=W0221
+    def save(self, *args, **kwargs):
+        existing_instance_count = EcjuQuery.objects.filter(id=self.id).count()
+
+        # Only create a notification when saving a ECJU query for the first time
+        if existing_instance_count == 0:
+            super(EcjuQuery, self).save(*args, **kwargs)
```
Comment (BenAndrew-Kainos) [inline cases/models.py]: Try and move this line to after the if/else to reduce duplication. If this doesn't work please leave a comment to why this needs to happen before the other statements in your if

File: cases/views.py [hunk 7]
```diff
@@ -268,15 +269,21 @@ def post(self, request, pk):
...
         """
+        if isinstance(request.user, ExporterUser):
+            serializer = EcjuQueryExporterSerializer
+        else:
+            serializer = EcjuQueryGovSerializer
+
         case = get_case(pk)
         case_ecju_queries = EcjuQuery.objects.filter(case=case)
-        serializer = EcjuQuerySerializer(case_ecju_queries, many=True)
+        serializer = serializer(case_ecju_queries, many=True)
```
Comment (BenAndrew-Kainos) [inline cases/views.py]: could this be named better than serializer = serializer()

File: users/models.py [hunk 8]
```diff
@@ -76,11 +76,13 @@ def __str__(self):
 
     objects = CustomUserManager()
 
+    def send_notification(self, case_note=None, ecju_query=None):
+        from cases.models import Notification
```
Comment (BenAndrew-Kainos) [inline users/models.py]: Does this import need to be here?

File: users/models.py [hunk 9]
```diff
@@ -76,11 +76,13 @@ def __str__(self):
...
+        Notification.objects.create(user=self, case_note=case_note, ecju_query=ecju_query)
```
Comment (BenAndrew-Kainos) [inline users/models.py]: Is it acceptable to create a Notification through this method with a None case_note and a None ecju_query?

File: users/models.py [hunk 10]
```diff
@@ -76,11 +76,13 @@ def __str__(self):
...
+
 
 class ExporterUser(BaseUser):
-    def send_notification(self, case):
```
Comment (BenAndrew-Kainos) [inline users/models.py]: Is ExporterUser redundant now if all it does is inherit BaseUser? Can this be removed?

File: users/models.py (hunk 8 above)
Comment (Mark-Spiers-kainos) [reply users/models.py]: circular import prevention

Comment (BenAndrew-Kainos) [reply users/models.py]: Ok please leave a comment mentioning that

File: users/models.py (hunk 10 above)
Comment (Mark-Spiers-kainos) [reply users/models.py]: still required

**Pull request merged**